new(mc): operatingsystems linux snmpv3 (#2891)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmpv3.md

| Metric name | Unit  |
|:------------|:------|
| nbproc      | N/A   |
| mem_total   | B     |
| mem_avg     | B     |
| cpu_total   | %     |

</TabItem>
<TabItem value="Swap" label="Swap">

| Metric name           | Unit  |
|:----------------------|:------|
| swap.usage.bytes      | B     |
| swap.free.bytes       | B     |
| swap.usage.percentage | %     |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Tcpcon-Generic" label="Tcpcon-Generic">

| Metric name                       | Unit  |
|:----------------------------------|:------|
| service.connections.tcp.count     | count |
| application.connections.tcp.count | count |

</TabItem>
<TabItem value="Uptime" label="Uptime">

| Metric name           | Unit  |
|:----------------------|:------|
| system.uptime.seconds | s     |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
</Tabs>

## Prerequisites

To monitor a Linux-based device, the SNMP service must be installed and configured. Most Linux distributions rely on net-snmp. 

## net-snmp server configuration

> A detailed documentation on how to configure SNMP is available in the documentation of each Linux distribution.
> 
> Here is a way to configure a SNMP v3 user:

Author: ```suggestion
**Notice:** The commands listed below may change depending on the distribution. A detailed documentation on how to configure SNMP is available in the documentation of each Linux distribution.

Here is a way to configure a SNMP v3 user that should work on most major distibutions.

Log in with SSH on the host to monitor and run the following commands:

```

> Here is a way to configure a SNMP v3 user:

Author: In a root shell

```
systemctl stop snmpd.service
net-snmp-create-v3-user -ro -A centreonrocks -X linuxisgreat -a SHA -x AES centreon
systemctl start snmpd.service
```

> The followin lines should be displayed as a return:

Reviewer: ```suggestion
The following lines should be returned:
```

5. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The host appears in the list of hosts, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the host: it shows the values of the macros.

### Using a service template provided by the connector

1. If you have used a host template and checked **Create Services linked to the Template too**, the services linked to the template have been created automatically, using the corresponding service templates. Otherwise, [create manually the services you want](/docs/monitoring/basic-objects/services) and apply a service template to them.
2. Fill in the macros you want (e.g. to change the thresholds for the alerts). Some macros are mandatory (see the table below).

<Tabs groupId="sync">
<TabItem value="Cpu" label="Cpu">

| Macro        | Description                                                                                        | Default value     | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | WARNING      | Warning threshold average CPU utilization                                                          | 80                |             |

Reviewer: ```suggestion
| WARNING      | Warning threshold for average CPU utilization                                                          | 80                |             |
```

> | CRITICAL     | Critical threshold average CPU utilization                                                         | 90                |             |

Reviewer: ```suggestion
| CRITICAL     | Critical threshold for average CPU utilization                                                         | 90                |             |
```

| WARNINGIDLE  | Warning threshold in percent                                                                       | 20:               |             |
| CRITICALIDLE | Critical threshold in percent                                                                      | 10:               |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose         |             |

</TabItem>
<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">

| Macro        | Description                                                                                                                                                                                    | Default value                                 | Mandatory   |
|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|

> | DISKNAME     | Set the storage (number expected) example: 1, 2,... (empty means 'check all storage')                                                                                                          |                                               |             |

Reviewer: ```suggestion
| DISKNAME     | Define the storage units to be checked (number expected) example: 1, 2,... (empty means 'check all storage units').                                                                                                         |                                               |             |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                                                             | --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-Global" label="Disk-Global">

| Macro        | Description                                                                                                                                                                                    | Default value                                           | Mandatory   |
|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|

> | FILTER       | Set the storage (number expected) example: 1, 2,... (empty means 'check all storage')                                                                                                          | .*                                                      |             |

Reviewer: ```suggestion
| FILTER       | Define the storage units to be checked (number expected) example: 1, 2,... (empty means 'check all storage units').                                                                                                          | .*                                                      |             |
```

| CRITICALREAD  | Critical threshold                                                                                 |                   |             |
| WARNINGWRITE  | Warning threshold                                                                                  |                   |             |
| CRITICALWRITE | Critical threshold                                                                                 |                   |             |
| EXTRAOPTIONS  | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Inodes-Global" label="Inodes-Global">

| Macro        | Description                                                                                        | Default value                                            | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------|:---------------------------------------------------------|:-----------:|

> | FILTER       | Set the disk path (number expected) example: 1, 2,... (empty means 'check all disks path')         | .*                                                       |             |

Reviewer: ```suggestion
| FILTER       | Specify the path of the disk you want to check (number expected) example: 1, 2,... (empty means 'check all disk paths')        | .*                                                       |             |
```

|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| WARNING      | Thresholds                                                                                         | 80                |             |
| CRITICAL     | Thresholds                                                                                         | 90                |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Macro        | Description                                                                                                     | Default value     | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| NTPADDR      | Set the ntp hostname (if not set, localtime is used)                                                            |                   |             |

> | NTPPORT      | Set the ntp port (default: 123)                                                                                 |                   |             |

Reviewer: ```suggestion
| NTPPORT      | Define the port of the NTP server (default: 123)                                                                                 |                   |             |
```

> | TIMEZONE     | Set the timezone of distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100' |                   |             |

Reviewer: ```suggestion
| TIMEZONE     | Set the timezone of the distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100' |                   |             |
```

| WARNING      | Time offset warning threshold (in seconds)                                                                      | -1:1              |             |

> | CRITICAL     | Time offset critical Threshold (in seconds)                                                                     | -2:2              |             |

Reviewer: ```suggestion
| CRITICAL     | Time offset critical threshold (in seconds)                                                                     | -2:2              |             |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).              |                   |             |

</TabItem>
<TabItem value="Process-Generic" label="Process-Generic">

| Macro        | Description                                                                                        | Default value     | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | PROCESSNAME  | Filter process name                                                                                |                   |             |
> | PROCESSPATH  | Filter process path                                                                                |                   |             |
> | PROCESSARGS  | Filter process arguments                                                                           |                   |             |

Author: ```suggestion
| PROCESSNAME  | Regular expression to define which processes should be monitored based on the executable’s name.  |                   |             |
| PROCESSPATH  | Regular expression to define which processes should be monitored based on the executable’s path.  |                   |             |
| PROCESSARGS  | Regular expression to define which processes should be monitored based on the command’s arguments.  |                   |             |
```

| --warning-access                                | Warning threshold.                                                                                                                                                                                                                            |
| --critical-access                               | Critical threshold. Check if storage is readOnly: --critical-access=readOnly                                                                                                                                                                  |
| --add-access                                    | Check storage access (readOnly, readWrite).                                                                                                                                                                                                   |
| --units                                         | Units of thresholds (default: '%') ('%', 'B').                                                                                                                                                                                                |

> | --free                                          | Thresholds are on free space left.                                                                                                                                                                                                            |

Reviewer: ?

> | --free                                          | Thresholds are on free space left.                                                                                                                                                                                                            |

Author: ```suggestion
| --free                                          | Apply thresholds on free space instead of used space.                                                                                                                                                                                                            |
```

> | --storage                                       | Set the storage (number expected) example: 1, 2,... (empty means 'check all storage').                                                                                                                                                        |

Reviewer: ```suggestion
| --storage                                       | Define the storage units to be checked (number expected) example: 1, 2,... (empty means 'check all storage units').                                                                                                                                                       |
```

| --statefile-key        | Define the key to encrypt/decrypt the cache.                                                                                                                                                                                                  |
| --statefile-cipher     | Define the cipher algorithm to encrypt the cache (default: 'AES').                                                                                                                                                                            |
| --warning-*            | Warning threshold. Can be: 'read', 'write', 'read-iops', 'write-iops', 'total-read', 'total-write', 'total-read-iops', 'total-write-iops', 'sum-read-write', 'sum-read-write-iops'.                                                           |
| --critical-*           | Critical threshold. Can be: 'read', 'write', 'read-iops', 'write-iops', 'total-read', 'total-write', 'total-read-iops', 'total-write-iops', 'sum-read-write', 'sum-read-write-iops'.                                                          |

> | --device               | Set the device (number expected) example: 1, 2,... (empty means 'check all devices').                                                                                                                                                         |

Reviewer: ```suggestion
| --device               | Define the device you want to check (number expected) example: 1, 2,... (empty means 'check all devices')                                                                                                                                                     |
```

> | --name                 | Allows to use device name with option --device instead of devoceoid index.                                                                                                                                                                    |

Reviewer: ```suggestion
| --name                 | Allows to use device name with option --device instead of device OID index.                                                                                                                                                                    |
```

</TabItem>
<TabItem value="Inodes-Global" label="Inodes-Global">

| Option                                          | Description                                                                                                                                                                                        |
|:------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --warning-usage                                 | Warning threshold in percent.                                                                                                                                                                      |
| --critical-usage                                | Critical threshold in percent.                                                                                                                                                                     |

> | --diskpath                                      | Set the disk path (number expected) example: 1, 2,... (empty means 'check all disks path').                                                                                                        |

Reviewer: ```suggestion
| --diskpath                                      | Specify the path of the disk you want to check (number expected) example: 1, 2,... (empty means 'check all disk paths').                                                                                                        |
```

> | --name                                          | Allows to use disk path name with option --diskpath instead ofdisk path oid index.                                                                                                                 |

Reviewer: ```suggestion
| --name                                          | Allows to use disk path name with option --diskpath instead of disk path OID index.                                                                                                                 |
```

| --units-cast                                    | Units of thresholds for communication types (default: 'percent\_delta') ('percent\_delta', 'percent', 'delta', 'deltaps', 'counter').                                                                                                                                                      |
| --nagvis-perfdata                               | Display traffic perfdata to be compatible with nagvis widget.                                                                                                                                                                                                                              |
| --interface                                     | Set the interface (number expected) example: 1,2,... (empty means 'check all interfaces').                                                                                                                                                                                                 |

> | --name                                          | Allows you to define the interface (in option --interface) byname instead of OID index. The name matching mode supports regular expressions.                                                                                                                                               |

Reviewer: ```suggestion
| --name                                          | Allows you to define the interface (in option --interface) by name instead of OID index. The name matching mode supports regular expressions.                                                                                                                                               |
```

| --speed-in                                      | Set interface speed for incoming traffic (in Mb).                                                                                                                                                                                                                                          |
| --speed-out                                     | Set interface speed for outgoing traffic (in Mb).                                                                                                                                                                                                                                          |
| --map-speed-dsl                                 | Get interface speed configuration for interface type 'adsl' and 'vdsl2'.  Syntax: --map-speed-dsl=interface-src-name,interface-dsl-name  Example: --map-speed-dsl=Et0.835,Et0-vdsl2                                                                                                            |

> | --force-counters64                              | Force to use 64 bits counters only. Can be used to improve performance.                                                                                                                                                                                                                    |

Reviewer: ```suggestion
| --force-counters64                              | Force to use 64-bit counters only. Can be used to improve performance.                                                                                                                                                                                                                    |
```

> | --force-counters32                              | Force to use 32 bits counters (even in snmp v2c and v3). Should be used when 64 bits counters are buggy.                                                                                                                                                                                   |

Reviewer: ```suggestion
| --force-counters32                              | Force to use 32-bit counters (even in snmp v2c and v3). Should be used when 64-bit counters are buggy.                                                                                                                                                                                   |
```

| --warning  | Warning threshold (1min,5min,15min).    |
| --critical | Critical threshold (1min,5min,15min).   |
| --average  | Load average for the number of CPUs.    |

</TabItem>
<TabItem value="Memory" label="Memory">

| Option                   | Description                                                                                                                                                                                                                                                                                          |
|:-------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| --units                  | Units of thresholds (default: '%') ('%', 'absolute')(deprecated. Please use new counters directly)                                                                                                                                                                                                   |

> | --free                   | Thresholds are on free space left (deprecated. Please use newcounters directly)                                                                                                                                                                                                                      |

Reviewer: ?

> | --free                   | Thresholds are on free space left (deprecated. Please use newcounters directly)                                                                                                                                                                                                                      |

Author: ```suggestion
| --free                   | Apply the thresholds on free space instead of on used space. This option is deprecated, please use --(warning|critical)-usage-free or  --(warning|critical)-swap-free instead. |
```

| --patch-redhat           | If using RedHat distribution with net-snmp \>= 5.7.2-43 and net-snmp \< 5.7.2-47. But you should update net-snmp!!!!  This version: used = memTotalReal - memAvailReal // free = memAvailReal  Others versions: used = memTotalReal - memAvailReal - memBuffer - memCached // free = total - used    |

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Option            | Description                                                                                                         |
|:------------------|:--------------------------------------------------------------------------------------------------------------------|
| --oid             | Override default OID.                                                                                               |
| --warning-offset  | Time offset warning threshold (in seconds).                                                                         |

> | --critical-offset | Time offset critical Threshold (in seconds).                                                                        |

Reviewer: ```suggestion
| --critical-offset | Time offset critical threshold (in seconds).                                                                        |
```

> | --ntp-hostname    | Set the ntp hostname (if not set, localtime is used).                                                               |

Reviewer: ```suggestion
| --ntp-hostname    | Set the NTP server's hostname (if not set, localtime is used)                                                               |
```

> | --ntp-port        | Set the ntp port (default: 123).                                                                                    |

Reviewer: ```suggestion
| --ntp-port        | Define the port of the NTP server (default: 123).                                                                                    |
```

> | --timezone        | Set the timezone of distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100'.    |

Reviewer: ```suggestion
| --timezone        | Set the timezone of the distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100'.    |
```

| --regexp-args          | Allows to use regexp to filter process arguments (with option --process-args).                                                                                                                                                                |
| --warning              | Warning threshold of matching processes count.                                                                                                                                                                                                |
| --critical             | Critical threshold of matching processes count.                                                                                                                                                                                               |
| --memory               | Check memory usage.                                                                                                                                                                                                                           |

> | --warning-mem-each     | Warning threshold of memory used by each matching processes (in Bytes).                                                                                                                                                                       |

Reviewer: ```suggestion
| --warning-mem-each     | Warning threshold of memory used by each matching process (in Bytes).                                                                                                                                                                       |
```

> | --critical-mem-each    | Critical threshold of memory used by each matching processes (in Bytes).                                                                                                                                                                      |

Reviewer: ```suggestion
| --critical-mem-each    | Critical threshold of memory used by each matching process (in Bytes).                                                                                                                                                                      |
```

| --warning-mem-avg      | Warning threshold of average memory used by matching processes (in Bytes).                                                                                                                                                                    |
| --critical-mem-avg     | Critical threshold of average memory used by matching processes (in Bytes).                                                                                                                                                                   |
| --cpu                  | Check cpu usage. Should be used with fix processes. If processespid changes too much, the plugin can't compute values.                                                                                                                        |
| --warning-cpu-total    | Warning threshold of cpu usage for all processes (in percent). CPU usage is in % of one cpu, so maximum can be 100% * number of CPU and a process can have a value greater than 100%.                                                         |

> | --critical-cpu-total   | Critical threshold of cpu usage for all processes (in percent). CPU usage is in % of one cpu, so maximum can be 100% * number of CPU and a process can have a value greater than 100%.                                                        |

Reviewer: ```suggestion
| --critical-cpu-total   | Critical threshold of CPU usage for all processes (in percent). CPU usage is in % of one CPU, so maximum can be 100% * number of CPU and a process can have a value greater than 100%.                                                        |
```

| --warning-uptime       | Warning threshold.                                                                                                                                                                                                                            |
| --critical-uptime      | Critical threshold.                                                                                                                                                                                                                           |
| --add-sysdesc          | Display system description.                                                                                                                                                                                                                   |
| --force-oid            | Can choose your oid (numeric format only).                                                                                                                                                                                                    |

> | --check-overload       | Uptime counter limit is 4294967296 and overflow. With that option, we manage the counter going back. But there is a few chance we can miss a reboot.                                                                                          |

Reviewer: ?

> | --check-overload       | Uptime counter limit is 4294967296 and overflow. With that option, we manage the counter going back. But there is a few chance we can miss a reboot.                                                                                          |

Author: ```suggestion
| --check-overload       | The uptime counter limit is 4294967296 centiseconds (~497 days). This option makes the plugin store the actual uptime by incrementing the number of revolutions of the counter when it resets to zero. |
```

---
id: operatingsystems-linux-snmpv3

> title: Linux SNMP

Author: ```suggestion
title: Linux SNMP v3
```
